### PR TITLE
Add Auto detect strategy to clearFormat API

### PR DESCRIPTION
**Pull request by Lufedi**
### Description

Implement auto detect clear format strategy.
Now we can pass a parameter to the `clearFormat` API  to specify the formatting strategy.:
* *Inline*: clear text format of inline elements
* *Block*: clear text and structure format of the selected elements
* *Autodetect*: detect what formatting strategy to use (Inline or Block) based on the current selection.

Example of the Autodetect strategy.

autodetectformat

### CODE REVIEW

File: demo/scripts/controls/ribbon/ribbonButtons.ts [hunk 1]
```diff
@@ -302,10 +302,18 @@ const buttons: { [key: string]: RibbonButtonType } = {
...
+                block: FormatType.Block,
+                selection: FormatType.Inline,
+                autodetect: FormatType.AutoDetect,
+            };
+            clearFormat(editor, handlers[key]);
+        },
         dropDownItems: {
             selection: 'Remove formatting of selected text',
             block: 'Remove formatting of selected paragraphs',
+            autodetect: 'Remove format (Autodetect)',
```
Comment (JiuqingSong) [inline demo/scripts/controls/ribbon/ribbonButtons.ts:316]: Let's put this as first menu item

File: packages/roosterjs-editor-api/lib/format/clearBlockFormat.ts [hunk 2]
```diff
@@ -23,6 +23,8 @@ const ATTRIBUTES_TO_PRESERVE = ['href', 'src'];
 const TAGS_TO_STOP_UNWRAP = ['TD', 'TH', 'TR', 'TABLE', 'TBODY', 'THEAD'];
 
 /**
+ * @deprecated
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-api/lib/format/clearBlockFormat.ts:26]: Also mention we which function and what parameter we should use instead

File: packages/roosterjs-editor-api/lib/format/clearBlockFormat.ts [hunk 3]
```diff
@@ -23,6 +23,8 @@ const ATTRIBUTES_TO_PRESERVE = ['href', 'src'];
 const TAGS_TO_STOP_UNWRAP = ['TD', 'TH', 'TR', 'TABLE', 'TBODY', 'THEAD'];
 
 /**
+ * @deprecated
+ * Use clearFormat passing the FormatType.Autodetect as parameter
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-api/lib/format/clearBlockFormat.ts:27]: please move the implementation of this function into clearFormat, and call into that function so this function becomes almost empty

File: packages/roosterjs-editor-api/lib/format/clearFormat.ts [hunk 4]
```diff
@@ -10,13 +12,61 @@ import { ChangeSource, DocumentCommand, IEditor, QueryScope } from 'roosterjs-ed
 
 const STYLES_TO_REMOVE = ['font', 'text-decoration', 'color', 'background'];
 
+export const enum FormattingStrategy {
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-api/lib/format/clearFormat.ts:15]: Move this type decaration into roosterjs-editor-types package, and use a better name to show it is for clear format only

File: packages/roosterjs-editor-api/lib/format/clearFormat.ts [hunk 5]
```diff
@@ -70,3 +120,23 @@ export default function clearFormat(editor: IEditor) {
...
+ * changed to default format. The format that get cleaned include B/I/U/font name/
+ * font size/text color/background color/align left/align right/align center/superscript/subscript
+ * @param editor The editor instance
+ * @param formatType type of format to apply
+ */
+export default function clearFormat(
+    editor: IEditor,
+    formatType: FormattingStrategy = FormattingStrategy.Inline
+) {
+    if (formatType == FormattingStrategy.Inline) {
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-api/lib/format/clearFormat.ts:135]: Use switch-case

File: packages/roosterjs-editor-api/lib/format/clearBlockFormat.ts (hunk 3 above)
Comment (Lufedi) [reply packages/roosterjs-editor-api/lib/format/clearBlockFormat.ts:27]: Alright

File: packages/roosterjs-editor-api/lib/format/clearFormat.ts (hunk 4 above)
Comment (Lufedi) [reply packages/roosterjs-editor-api/lib/format/clearFormat.ts]: Alright, I'll use ClearFormatMode

**Review by JiuqingSong — APPROVED** [review]

**Pull request merged**